GitUI/UserControls/RevisionGrid/Graph/LaneInfoProvider.cs

```diff
@@ -0,0 +1,193 @@
+                    var nodeFound = false;
+                    for (var nodeIndex = descJunction.NodeCount - 1; nodeIndex > 0; --nodeIndex)
+                    {
+                        var innerNode = descJunction[nodeIndex];
+                        if (nodeFound)
+                        {
+                            if (FindBranch(innerNode, node, descJunction))
+                            {
+                                return true;
+                            }
+                        }
+                        else
+                        {
+                            nodeFound = innerNode == node;
+                        }
+                    }
+
+                    // handle the youngest and its descendants
+                    if (AddReferencesOf(descJunction.Youngest, descJunction))
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            private bool FindBranch([NotNull] Node descNode, [NotNull] Node node, [CanBeNull] Junction descJunction)
+            {
+                foreach (var gitReference in descNode.Revision.Refs)
+                {
+                    if (gitReference.IsHead || gitReference.IsRemote)
+                    {
+                        CheckForMerge(node, descJunction);
+                        CommittedTo = CommittedTo ?? gitReference.Name;
+                        return true;
+                    }
+
+                    if (gitReference.IsStash && CommittedTo == null)
+                    {
+                        CommittedTo = gitReference.Name;
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            /// <summary>
+            /// Checks whether the commit message is a merge message
+            /// and then if its a merge message, sets CommittedTo and MergedWith.
+            ///
+            /// MergedWith is set if it is the current node, i.e. on the first call.
+            /// MergedWith is set to string.Empty if it is no merge.
+            /// First/second branch does not matter because it is the message of the current node.
+            /// </summary>
+            /// <param name="node">the node of the revision to evaluate</param>
+            /// <param name="descJunction">
+            /// the descending junction the node is part of
+            /// (used for the decision whether the node belongs the first or second branch of the merge)
+            /// </param>
+            private bool CheckForMerge([NotNull] Node node, [CanBeNull] Junction descJunction)
+            {
+                var isTheFirstBranch = descJunction == null || node.Ancestors.Count == 0 || node.Ancestors.First() == descJunction;
+                string mergedInto;
+                string mergedWith;
+                (mergedInto, mergedWith) = ParseMergeMessage(node, isTheFirstBranch);
```

Comment:
@RussKie, why did you remove `appendPullRequest: ` from the argument `isTheFirstBranch` given in the call of `ParseMergeMessage`?

Reply:
I don't remember tbh, I suspect it may have been Re# cleaning the code.